renderer/FrameBuffer.h

```diff
@@ -46,3 +46,58 @@ void FB_BindShadowTexture();
 void FB_SelectPrimary();
 void FB_SelectPostProcess();
 void FB_ResolveMultisampling( GLbitfield mask = GL_COLOR_BUFFER_BIT, GLenum filter = GL_NEAREST );
+
+class FrameBuffer {
+public:
+	~FrameBuffer();
+
+	static FrameBuffer * Create( GLuint width, GLuint height, int msaa = 0 );
+	static FrameBuffer * BackBuffer();
+
+	void CreateColorBuffer();
+	void CreateDepthStencilBuffer();
+
+	void AddColorTexture( idImage *colorTexture );
+	void AddDepthStencilTexture( idImage *depthStencilTexture );
+	void AddDepthStencilTextures( idImage *depthTexture, idImage *stencilTexture );
+
+	void Validate();
+
+	void Bind();
+	void BindDraw();
+	void BindRead();
+
+	void BlitFullTo( FrameBuffer *target, GLbitfield mask = GL_COLOR_BUFFER_BIT, GLenum filter = GL_NEAREST );
+
+	GLuint GetWidth() const { return width; }
+
+	GLuint GetHeight() const { return height; }
+
+private:
+	FrameBuffer( GLuint fbo, GLuint width, GLuint height, int msaa );
+
+	GLuint fbo;
+	GLuint colorBuffer;
+	GLuint depthStencilBuffer;
+
+	GLuint width;
+	GLuint height;
+	int msaa;
+};
+
+struct frameBuffers_t {
+	FrameBuffer *primary;
+	FrameBuffer *resolve;
+	FrameBuffer *lightgem;
+	FrameBuffer *backBuffer;
+	FrameBuffer *finalOutput;
+
+	FrameBuffer *currentDraw;
+	FrameBuffer *currentRead;
+
+	std::vector<FrameBuffer*> allFrameBuffers;
```

Comment:
I suggest using idLib instead of STL in renderer.
Given that renderer is mostly written in pure C, idLib fits much better: it is very simple an lightweight.

Reply:
On the other hand, the STL has come a long way since id created idLib, and I believe I've seen Carmack mention that he was not overly happy with idLib. Modern vector implementations probably make use of C++1x features like move etc, giving better performance.

Of course, for this use case it's pretty much irrelevant, and I don't really have a strong preference.  I'll happily switch to idLib::List.